x-pack/plugins/ml/public/embeddables/anomaly_explorer/anomaly_explorer_setup_flyout.tsx

```diff
@@ -0,0 +1,115 @@
+import { getDefaultExplorerChartsPanelTitle } from './anomaly_explorer_embeddable';
+import { getMlGlobalServices } from '../../application/app';
+import { HttpService } from '../../application/services/http_service';
+import { DashboardConstants } from '../../../../../../src/plugins/dashboard/public';
+import { AnomalyExplorerEmbeddableInput } from '..';
+
+export async function resolveAnomalyExplorerUserInput(
+  coreStart: CoreStart,
+  input?: AnomalyExplorerEmbeddableInput
+): Promise<Partial<AnomalyExplorerEmbeddableInput>> {
+  const {
+    http,
+    uiSettings,
+    overlays,
+    application: { currentAppId$ },
+  } = coreStart;
+
+  const anomalyDetectorService = new AnomalyDetectorService(new HttpService(http));
+
+  return new Promise(async (resolve, reject) => {
+    const maps = {
+      groupsMap: getInitialGroupsMap([]),
+      jobsMap: {},
+    };
+
+    const tzConfig = uiSettings.get('dateFormat:tz');
+    const dateFormatTz = tzConfig !== 'Browser' ? tzConfig : moment.tz.guess();
+
+    const selectedIds = input?.jobIds;
+
+    const flyoutSession = coreStart.overlays.openFlyout(
+      toMountPoint(
+        <KibanaContextProvider services={{ ...coreStart, mlServices: getMlGlobalServices(http) }}>
+          <JobSelectorFlyoutContent
+            selectedIds={selectedIds}
+            withTimeRangeSelector={false}
+            dateFormatTz={dateFormatTz}
+            singleSelection={false}
+            timeseriesOnly={true}
+            onFlyoutClose={() => {
+              flyoutSession.close();
+              reject();
+            }}
+            onSelectionConfirmed={async ({ jobIds, groups }) => {
+              const title = input?.title ?? getDefaultExplorerChartsPanelTitle(jobIds);
+
+              const jobs = await anomalyDetectorService.getJobs$(jobIds).toPromise();
+
+              const influencers = anomalyDetectorService.extractInfluencers(jobs);
+              influencers.push(VIEW_BY_JOB_LABEL);
+
+              await flyoutSession.close();
+
+              const modalSession = overlays.openModal(
+                toMountPoint(
+                  <AnomalyExplorerInitializer
+                    defaultTitle={title}
+                    initialInput={input}
+                    onCreate={({ panelTitle, maxSeriesToPlot }) => {
+                      modalSession.close();
+
+                      resolve({
+                        jobIds,
+                        title: panelTitle,
+                        maxSeriesToPlot,
+                      });
+                    }}
+                    onCancel={() => {
+                      modalSession.close();
+                      reject();
+                    }}
+                  />
+                )
+              );
+            }}
+            maps={maps}
+          />
+        </KibanaContextProvider>
+      ),
+      {
+        'data-test-subj': 'mlFlyoutJobSelector',
+        ownFocus: true,
+        closeButtonAriaLabel: 'jobSelectorFlyout',
```

Comment:
`jobSelectorFlyout` will be the exact name used for the close button which isn't very clear.

The default is `Close this dialog`. If you'd like to improve that, `Close job selection dialog` could be a good improvement.

If you could fix this same issue in `x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_setup_flyout.tsx` that would be really great.

```suggestion
        closeButtonAriaLabel: 'Close job selection dialog', // but remember to i18n
```

Reply:
Updated here [`cb43b45` (#94396)](https://github.com/elastic/kibana/pull/94396/commits/cb43b4560e8dc54fe35aa2a9efffdd11b5c559e3)